go-controller/pkg/node/node.go

```diff
@@ -250,10 +252,45 @@ func (n *OvnNode) Start() error {
 	if !ok {
 		return fmt.Errorf("Cannot get kubeclient for starting CNI server")
 	}
+	err = n.WatchEndpoints()
+	if err != nil {
+		return err
+	}
 
 	// start the cni server
 	cniServer := cni.NewCNIServer("", kclient.KClient)
 	err = cniServer.Start(cni.HandleCNIRequest)
 
 	return err
 }
+
+func (n *OvnNode) WatchEndpoints() error {
+	_, err := n.watchFactory.AddEndpointsHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(old, new interface{}) {
+			epNew := new.(*kapi.Endpoints)
+			epOld := old.(*kapi.Endpoints)
+			if reflect.DeepEqual(epNew.Subsets, epOld.Subsets) {
+				return
+			}
+			newEndpointAddressMap := buildEndpointAddressMap(epNew.Subsets)
+			for _, subset := range epOld.Subsets {
+				for _, address := range subset.Addresses {
+					if _, ok := newEndpointAddressMap[address.IP]; ok {
```

Comment:
if ok means that the IP still exists, should not delete the ones that does not exist?
```suggestion
					if _, ok := newEndpointAddressMap[address.IP]; !ok {
```

Reply:
+1

Reply:
wow your right I missed that when I  changed this around. good spot 

done